src/js/Tables/OverflowTable.js

```diff
@@ -0,0 +1,469 @@
+			}
+		}
+
+		// Add other event listeners to update controls.
+		this.wrapper.addEventListener('scroll', updateScoll);
+		window.addEventListener('resize', updateScoll);
+		window.addEventListener('load', updateScoll);
+		this._listeners.push({ element: this.wrapper, updateScoll, type: 'scroll' });
+		this._listeners.push({element: window, updateScoll, type: 'resize'});
+		this._listeners.push({element: window, updateScoll, type: 'load'});
+	}
+
+	/**
+	 * Add hide/show functionality for long tables.
+	 * @returns {undefined}
+	 */
+	_setupExpander() {
+		if (!this.canExpand()) {
+			return;
+		}
+
+		if (!this.container || !this.wrapper) {
+			throw new Error(
+				'Controls to expand/contract the table could not be added to "o-table" as it is missing markup.' +
+				'Please add the container and wrapper element according to the documentation https://registry.origami.ft.com/components/o-table.'
+			);
+		}
+
+		// Add table controls (e.g. "more" button).
+		if (!this.controls) {
+			this._addControlsToDom();
+		}
+
+		if (this.controls.moreButton) {
+			const toggleExpanded = function () {
+				if (this.isExpanded()) {
+					this.contractTable();
+				} else {
+					this.expandTable();
+				}
+			}.bind(this);
+			this.controls.moreButton.addEventListener('click', toggleExpanded);
+			this._listeners.push({element: this.controls.moreButton, toggleExpanded, type: 'click'});
+		}
+
+		if (this._opts.expanded){
+			this.expandTable();
+		} else {
+			this.contractTable();
+		}
+	}
+
+	/**
+	 * Set table scroll position in wrapper.
+	 * @returns {undefined}
+	 */
+	_setScrollPosition() {
+		this._fromEnd = this.wrapper.scrollWidth - this.wrapper.clientWidth - this.wrapper.scrollLeft;
+		this._fromStart = this.wrapper.scrollLeft;
+	}
+
+	/**
+	 * Update all controls and their overlays,
+	 * e.g. forward/back arrow visibility, visibility of arrow dock, overlay fade.
+	 * @returns {undefined}
+	 */
+	_updateControls() {
+		if (!this._controlUpdateScheduled) {
+			this._controlUpdateScheduled = true;
+			setTimeout(function () {
+				// Toggle fade.
+				this.controls.fadeOverlay.classList.toggle('o-table-fade-overlay--scroll', this._canScrollTable);
+				this.controls.fadeOverlay.style.setProperty('--o-table-fade-from-end', `${Math.min(this._fromEnd, 10)}px`);
+				this.controls.fadeOverlay.style.setProperty('--o-table-fade-from-start', `${Math.min(this._fromStart, 10)}px`);
+
+				// Toggle arrow dock.
+				this.controls.controlsOverlay.classList.toggle('o-table-control-overlay--arrow-dock', this._showArrowDock);
+
+				// Update forward/back scroll controls.
+				if (OverflowTable._supportsArrows()) {
+					this._updateScrollControl(this.controls.forwardButton);
+					this._updateScrollControl(this.controls.backButton);
+				}
+
+				// Make controls visible.
+				this.controls.controlsOverlay.style.display = '';
+				this.controls.fadeOverlay.style.display = '';
+				this._controlUpdateScheduled = false;
+			}.bind(this), 33);
```

Comment:
Quite arbitrary. Roughly 30fps. This could be better. 😞I didn't want to update more frequently as it's only relevant as you reach the start/end of the table. I could not figure out a way to use the intersection observer in this case, as the controls are a sticky child of the container and not of the scrollable wrapper.